tests/regression/run_harnesses.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Function to run a harness
+run_harness() {
+    local harness=$1
+    local output_file="cachegrind.out.${harness}"
+    local annotated_output="perf_outputs/${harness}_annotated.txt"
+
+    echo "Running harness: $harness"
+    cargo build
```

Comment:
I'd vote for removing the `cargo build` invocation from the `run_harnesses.sh` script. Shell Scripts are always hard to read, so I like to keep them as simple as possible.